dashboard/client/src/common/ServeStatus.component.test.tsx

```diff
@@ -37,33 +43,18 @@ describe("ServeStatusIcon", () => {
     );
   });
 
-  it("renders DEPLOY_FAILED status", () => {
+  it("renders DEPLOY_FAILED status", async () => {
     render(
       <ServeStatusIcon
         app={{ ...APP, status: ServeApplicationStatus.DEPLOY_FAILED }}
         small={false}
       />,
     );
+
+    await screen.findByTestId("serve-status-icon");
+
     const icon = screen.getByTestId("serve-status-icon");
     const classList = icon.getAttribute("class");
     expect(classList).toContain("colorError");
   });
-
-  it("renders DEPLOYING and DELETING status", () => {
```

Comment:
I just deleted the two test cases here because
1. we could not use findByTestId here since the icon did not exist
2. They are not necessary so it is ok to remove them

Reply:
Another approach is to use waitFor(() => new Promise((resolve) => setTimeout(resolve, 1000)));
to await the render to complete

Reply:
> I just deleted the two test cases here because
> 
> 1. we could not use findByTestId here since the icon did not exist
> 2. They are not necessary so it is ok to remove them

ok fine.



> Another approach is to use `waitFor(() => new Promise((resolve) => setTimeout(resolve, 1000)));` to await the render to complete

no, we should not do it this way. It intoduces flakiness (sometimes test pass sometimes test may fail) and also makes tests take longer than they should.

Reply:
Got it! I agree it would introduce flakiness since it is hard to determine the right amount of time to wait